platform/windows/detect.py

```diff
@@ -71,6 +71,7 @@ def get_opts():
         BoolVariable("use_mingw", "Use the Mingw compiler, even if MSVC is installed. Only used on Windows.", False),
         BoolVariable("use_llvm", "Use the LLVM compiler", False),
         BoolVariable("use_thinlto", "Use ThinLTO", False),
+        BoolVariable("use_static_cpp", "When using MinGW, link libgcc, libstdc++ and libpthread statically", True),
```

Comment:
Same option should control MSVC runtime linking as well.

See line 225 - `env.AppendUnique(CCFLAGS=["/MT", "/Gd", "/GR", "/nologo"])`, `/MT` for static `/MD` for dynamic.

[Run-Time library linking options.](https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-160)